### PR TITLE
allow only one stake pool to operate on a single node

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

**Review by Zeegomo — COMMENTED** [review]
Multiple BFT leaders can be supported without the need to have multiple pools. Not that it's a super common deployment but it does not incur in significant complexity on our side (except a bit of key management).
Also the capability of changing leader keys 'live' could be nice to retain anyway (for example returning an error if more than one praos keys are set at once)

**Comment by eugene-babichenko** [comment]
>  Also the capability of changing leader keys 'live' could be nice to retain anyway (for example returning an error if more than one praos keys are set at once)

Do you refer to the case where we left the possibility to have multiple BFT leaders here?

**Comment by Zeegomo** [comment]
> Do you refer to the case where we left the possibility to have multiple BFT leaders here?

It looks like this capability was removed in this PR.

I'm saying that this PR removes a bit more than just multiple stake pools on a single node like:
* multiple BFT leaders
* key management without having to restart the node (for both praos and bft keys)

And I'm not sure we should remove these

**Comment by Zeegomo** [comment]
Agree these are rather unusual features, but while I know why we introduced the multi-pool thing, I lack context for these other two. Maybe there's something I'm missing?

**Comment by eugene-babichenko** [comment]
The test failures are older ones from `master` and are fixed in https://github.com/input-output-hk/jormungandr/pull/3559

**Comment by mzabaluev** [comment]
Needs rebasing.

**Comment by mzabaluev** [comment]
To reiterate why we want to reduce to one stake pool/ BFT leader per node:

## Reasons to do this

* Saner implementation of mempool logic, fragment log, metrics, other APIs.
* Remove any potential unfairness from running leaders in the same node, leaving a flat network organization (leaving only connectivity/latency issues).

Anything else?

## What are we losing

* The use case of running a shared jormungandr node for multiple stake pools/consensus leaders, potentially segregating their secrets in separate enclave services. This is easily replaced by hosting a node per each pool, for the small price of duplicated storage. In practice, I expect networks would prefer horizontally scaling their leader node instances as much as possible anyway.

File: doc/api/v0.yaml
Comment (mzabaluev) [inline doc/api/v0.yaml:338]: Looks like this response payload has also changed, no `enclave_leader_id` any more?

File: jormungandr/src/fragment/pool.rs
Comment (mzabaluev) [inline jormungandr/src/fragment/pool.rs:35]: Rename to `Pool` for clarity?

File: jormungandr/src/settings/command_arguments.rs
Comment (mzabaluev) [inline jormungandr/src/settings/command_arguments.rs:28]: The doc needs to be updated.

**Review by mzabaluev — COMMENTED** [review]
Looks good. Minor changes need to be made and the doc updated: it still claims that multiple --secret options can be given.

**Comment by eugene-babichenko** [comment]
> What are we losing

We are also losing the capability to manage leader keys through the API. Although I think change of keys is a rare event and we do not benefit from having such functionality.

**Review by mzabaluev — APPROVED** [review]
Good to go after resolving the merge conflict.

**Pull request merged**
